src/main/java/edu/rosehulman/covidtracer/controller/PositiveCaseController.java

```diff
@@ -1,7 +1,108 @@
+import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
+import edu.rosehulman.covidtracer.model.ContactTracer;
+import edu.rosehulman.covidtracer.model.Person;
+import edu.rosehulman.covidtracer.model.PositiveCase;
+import edu.rosehulman.covidtracer.repository.ContactTracerRepository;
+import edu.rosehulman.covidtracer.repository.PersonRepository;
+import edu.rosehulman.covidtracer.repository.PositiveCaseRepository;
+import edu.rosehulman.covidtracer.service.PositiveCaseService;
+import javassist.NotFoundException;
+
+@CrossOrigin(origins = "http://localhost:4200", maxAge = 3600)
 @RestController
 public class PositiveCaseController {
+
+	@Autowired
+	private PositiveCaseService service;
+
+	@Autowired
+	private PositiveCaseRepository repository;
+
+	@Autowired
+	private PersonRepository personRepo;
+
+	@Autowired
+	private ContactTracerRepository contactTracerRepo;
+
+	@GetMapping("/positive-case")
+	public ResponseEntity<List<PositiveCase>> getAllPositiveCases(@RequestParam(defaultValue = "0") Integer pageNum,
+			@RequestParam(defaultValue = "10") Integer entriesPerPage,
+			@RequestParam(defaultValue = "ID") String sortBy) {
+		List<PositiveCase> resultSet = service.getAllPositiveCases(pageNum, entriesPerPage, sortBy);
+		return new ResponseEntity<List<PositiveCase>>(resultSet, new HttpHeaders(), HttpStatus.OK);
+	}
+
+	@GetMapping("/positive-case/{id}")
+	public PositiveCase retrievePositiveCase(@PathVariable Long id) throws NotFoundException {
+		Optional<PositiveCase> positiveCase = repository.findById(id);
+
+		if (!positiveCase.isPresent())
+			throw new NotFoundException("id-" + id);
+
+		return positiveCase.get();
+	}
+
+	@PostMapping("/userlist/{id}")
+	@ResponseStatus(HttpStatus.CREATED)
+	public ResponseEntity<List<Person>> createPositiveCaseOnPersonProfile(@PathVariable(value = "id") Long personID,
+			@RequestBody PositiveCase positiveCase) throws NotFoundException {
+		return personRepo.findById(personID).map(person -> {
+			positiveCase.setPerson(person);
+			repository.save(positiveCase);
+			List<Person> resultSet = new ArrayList<Person>();
+			resultSet.add(person);
+			return new ResponseEntity<List<Person>>(resultSet, new HttpHeaders(), HttpStatus.OK);
+		}).orElseThrow(() -> new NotFoundException("PersonID" + personID + " not found"));
+
+	}
+
+	@PostMapping("/userlist")
+	@ResponseStatus(HttpStatus.CREATED)
+	public PositiveCase createPositiveCaseOnPeopleList(@RequestParam(name = "id") Long personID,
+			@RequestBody PositiveCase positiveCase) throws NotFoundException {
+		return personRepo.findById(personID).map(person -> {
+			positiveCase.setPerson(person);
+			return repository.save(positiveCase);
+		}).orElseThrow(() -> new NotFoundException("PersonID" + personID + " not found"));
+
+	}
+	
+	@CrossOrigin(origins = "http://localhost:4200", maxAge = 3600)
+	@PutMapping("/positive-case/{id}")
```

Comment:
This is the endpoint to add a contact tracer?

Reply:
Yes.

Reply:
Yes.